cucumber-selenium-grid/README.md

```diff
@@ -0,0 +1,191 @@
+* [cucumber](https://cucumber.io/) as the interepreter of gherkin
+* [protractor](https://www.protractortest.org/#/) as the framework to run the tests (as this is an angularjs application)
+* a custom jenkins slave able to run the protractor tests
+* [zalenium](https://opensource.zalando.com/zalenium/) to dynamically create [selenium grid](https://www.seleniumhq.org/docs/07_selenium_grid.jsp) clusters of variable size based on demand in kubernetes
+* selenium hub, the brain of the selenium cluster
+* selenium nodes with chrome and firefox
+* the target application
+
+The following breaks down the architecture of the pipeline deployed, as well as walks through the manual deployment steps
+
+### OpenShift Templates
+
+The components of this pipeline are divided into two templates.
+
+The first template, `applier/templates/build.yml` is what we are calling the "Build" template. It contains:
+
+* A `jenkinsPipelineStrategy` BuildConfig
+* An `s2i` BuildConfig
+* An ImageStream for the s2i build config to push to
+
+The build template contains a default source code repo for a java application compatible with this pipelines architecture (https://github.com/redhat-cop/spring-rest).
+
+The second template, `applier/templates/deployment.yml` is the "Deploy" template. It contains:
+
+* A tomcat8 DeploymentConfig
+* A Service definition
+* A Route
+
+The idea behind the split between the templates is that I can deploy the build template only once (to my dev project) and that the pipeline will promote my image through all of the various stages of my application's lifecycle. The deployment template gets deployed once to each of the stages of the application lifecycle (once per OpenShift project).
+
+There is also an additional template that sets up the zalenium infrastructure in `applier/templates/selenium-grid.yaml`
+
+### Pipeline Script
+
+This project includes a sample `Jenkinsfile` pipeline script that could be included with a Java project in order to implement a basic CI/CD pipeline for that project, under the following assumptions:
+
+* The project is built with Maven
+* The OpenShift projects that represent the Application's lifecycle stages are of the naming format: `<app-name>-dev`, `<app-name>-stage`, `<app-name>-prod`.
+
+This pipeline defaults to use our [Spring Boot Demo App](https://github.com/redhat-cop/spring-rest).
+
+## Bill of Materials
+
+* One or Two OpenShift Container Platform Clusters
+  * OpenShift 3.5+ is required
+  * [Red Hat OpenJDK 1.8](https://access.redhat.com/containers/?tab=overview#/registry.access.redhat.com/redhat-openjdk-18/openjdk18-openshift) image is required
+* Access to GitHub
+* dynamic provisioning able to provsion RWO and RWM types of volumes.
+* ability to make calls from the dev project to the stage project
+
+## Manual Deployment Instructions
+
+### tl;dr
+```
+oc process -f applier/projects/projects.yml | oc apply -f -
```

Comment:
no instruction to create projects here. Also, in place of the tl;dr instructions, I would include the instructions for running the applier automation.

Reply:
not sure what you mean , that line creates the projects.
for the applier I'll need someone's help, there are a few things I wouldn't knot how to do, such as:
1. customizing Jenkins
2. modifying the scc of a service account.